src/Microsoft.Extensions.Configuration/IncludeConfigurationProvider.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.Configuration
+{
+    public class IncludeConfigurationProvider : ConfigurationProvider
+    {
+        public IncludeConfigurationProvider(IConfiguration source)
+        {
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+            foreach (var section in source.GetChildren())
+            {
+                AddSection(section, section.Key);
+            }
+        }
+
+        private void AddSection(IConfigurationSection section, string keyPrefix)
```

Comment:
Nit: this is not a `keyPrefix` but a `parentKey`. It helps given that we give "prefix" a different meaning elsewhere.